src/controller.rs

```diff
@@ -121,31 +111,37 @@ async fn reconcile_crd(config: &WasmCloudHostConfig, ctx: Arc<Context>) -> Resul
     let mut cfg = config.get(&name).await?;
     let secrets = Api::<Secret>::namespaced(kube_client, &ns);
 
-    let secret = secrets.get(&cfg.spec.secret_name).await.map_err(|e| {
-        warn!("Failed to read secrets: {}", e);
-        e
-    })?;
-    let s = Secrets::from_k8s_secret(&secret).map_err(|e| {
-        warn!("Failed to read secrets: {}", e);
-        Error::SecretError(format!(
-            "Failed to read all secrets from {}: {}",
-            secret.metadata.name.unwrap(),
+    let mut s = Secrets {
```

Comment:
not a fan of `s` here, however tried to minimize touch point

Reply:
I would rename this to `secrets` and call the line above ([#112](https://github.com/wasmCloud/wasmcloud-operator/pull/68/files#diff-1fd48ab302efbb91fe84507671007262db18094467e4ad707d036f60ff22e0eaR112)) `secrets_api` or something instead.